sqlmesh/core/engine_adapter/trino.py

```diff
@@ -176,11 +176,16 @@ def _df_to_source_queries(
         batch_size: int,
         target_table: TableName,
     ) -> t.List[SourceQuery]:
+        assert isinstance(df, pd.DataFrame)
```

Comment:
I added this because `df` is of type `DF`, which includes Spark dataframes, and for these `dtypes` [returns](https://spark.apache.org/docs/3.1.1/api/python/reference/api/pyspark.sql.DataFrame.dtypes.html#pyspark.sql.DataFrame.dtypes) a `list` which is not indexable by `str` (`column`). Should be fine, because `super()._df_to_source_queries` also makes this assertion.